converter/src/test/java/org/dswarm/converter/flow/test/json/JSONSourceResourceGDMStmtsFlowTest.java

```diff
@@ -0,0 +1,205 @@
+				.fromDataModel(dataModel, false);
+
+		testFlow(flow, "bib-record-marc.json", "test-record-marc-gdm.json", null);
+	}
+
+	private void testFlow(final JSONSourceResourceGDMStmtsFlow flow, final String fileName, final String expectedResultFileName, final Integer offset)
+			throws DMPConverterException {
+
+		final List<GDMModel> gdmModels = flow.applyResource(fileName).toList().toBlocking().first();
+
+		LOG.debug("GDM model = '{}'", gdmModels);
+
+		if (gdmModels != null && !gdmModels.isEmpty()) {
+
+			for (final GDMModel gdmModel : gdmModels) {
+
+				final Model model = gdmModel.getModel();
+
+				final JsonNode jsonNode = gdmModel.toJSON();
+
+				Assert.assertNotNull("the GDM model shouldn't be null", model);
+
+				// serialise model as JSON (pretty print)
+				final ObjectMapper mapper = DMPPersistenceUtil.getJSONObjectMapper().copy().configure(SerializationFeature.INDENT_OUTPUT, true);
+
+				String modelJSON = null;
+
+				try {
+
+					modelJSON = mapper.writeValueAsString(model);
+				} catch (final JsonProcessingException e) {
+
+					Assert.assertTrue("something went wrong while serialising the GDM model to JSON", false);
+				}
+
+				Assert.assertNotNull("the GDM model JSON shouldn't be null", modelJSON);
+
+				final URL expectedResultFileURL = Resources.getResource(expectedResultFileName);
+
+				Assert.assertNotNull("the expected result file URL shouldn't be null", expectedResultFileURL);
+
+				String expectedResult = null;
+
+				try {
+
+					expectedResult = Resources.toString(expectedResultFileURL, Charsets.UTF_8);
+				} catch (final IOException e) {
+
+					Assert.assertTrue("something went wrong while reading expected result from file", false);
+				}
+
+				Assert.assertNotNull("the expected result shouldn't be null", expectedResult);
+
+				final long expectedResultLength;
+
+				if (offset != null) {
+
+					// TODO http://data.slub-dresden.de/datamodels/
+
+					ArrayNode actualModelJSON = null;
+
+					try {
+
+						actualModelJSON = mapper.readValue(modelJSON, ArrayNode.class);
+					} catch (final JsonParseException e) {
+
+						Assert.assertTrue("something went wrong while deserializing the actual result", false);
+					} catch (final JsonMappingException e) {
+
+						Assert.assertTrue("something went wrong while deserializing the actual result", false);
+					} catch (final IOException e) {
+
+						Assert.assertTrue("something went wrong while deserializing the actual result", false);
+					}
+
+					Assert.assertNotNull("the deserialized result shouldn't be null", actualModelJSON);
+
+					final JsonNode firstElementJSON = actualModelJSON.get(0);
+
+					Assert.assertNotNull("the first element of the actual result JSON array shouldn't be null", firstElementJSON);
+					Assert.assertTrue("the first element of the actual result JSON array should be a JSON object",
+							firstElementJSON instanceof ObjectNode);
+
+					final ObjectNode firstElementJSONObject = (ObjectNode) firstElementJSON;
```

Comment:
In the same manner, I think these kind of checks can be simplified by just doing the cast. If its not the correct instance, it will fail and having a ClassCastException tells you just as much.